dm/syncer/ddl.go

```diff
@@ -1102,10 +1102,10 @@ func (ddl *DDLWorker) handleModifyColumn(qec *queryEventContext, info *ddlInfo,
 	switch {
 	case mysql.HasAutoIncrementFlag(oldCol.GetFlag()) && !mysql.HasAutoIncrementFlag(newCol.GetFlag()):
 		return bf.RemoveAutoIncrement, nil
-	case mysql.HasPriKeyFlag(oldCol.GetFlag()) != mysql.HasPriKeyFlag(newCol.GetFlag()):
```

Comment:
This part comes from the master branch, follows the same logic. 

Do you mean, if the col flag does not match, it may caused by `AddPrimaryKey` or `DropPrimaryKey`, but only the latter one covered here ?